### PR TITLE
Add ability for domain job status monitoring

### PR DESCRIPTION
**Pull request by vladiskuz**
Refactor a big part of code in object oriented style. Add ability for domain job status monitoring

### CODE REVIEW

File: migrationmonitor/actor.py [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
+class BaseActor(threading.Thread):
+
+    def __init__(self):
+        super(BaseActor, self).__init__()
+        self.daemon = True
+        self.q = Queue()
 
-    def tell(item):
-        q.put(item)
+    def add_task_to_queue(self, item):
```
Comment (rk4n) [inline migrationmonitor/actor.py]: it's actually `tell` in actors/akka terminology.
http://doc.akka.io/docs/akka/snapshot/java/untyped-actors.html#Tell__Fire-forget

File: migrationmonitor/actor.py [hunk 2]
```diff
@@ -1,32 +1,32 @@
...
-                fn(tell, item)
+                self._run(item)
             finally:
-                q.task_done()
+                self.q.task_done()
 
-    t = Thread(target=worker, name=fn.__name__)
-    t.daemon = True
-    t.start()
+    def _run(self, item):
```
Comment (rk4n) [inline migrationmonitor/actor.py]: As in previous line `on_receive` is better name for this method. As you actually react on some message.

File: migrationmonitor/watcher.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+import libvirt
+
+import actor
+import logger as log
+import settings
+import threading
+import utils
+
+
+class DomsWatcher(threading.Thread):
```
Comment (rk4n) [inline migrationmonitor/watcher.py]: `LibvirtDomainWatcher` is probably better name.

File: migrationmonitor/watcher.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+                known_dom_ids.remove(dom_id)
+                log.info(
+                    "Domain with id %s was not found on %s",
+                    dom_id,
+                    self.conn.getURI())
+
+            time.sleep(self.interval)
+
+
+class DomsJobMonitorActorCreator(threading.Thread):
```
Comment (rk4n) [inline migrationmonitor/watcher.py]: `DomainJob...` and it's actually can be actor activating once N seconds.

File: migrationmonitor/settings.py [hunk 5]
```diff
@@ -24,11 +23,3 @@
         "run": "",
     }
 }
-
```
Comment (rk4n) [inline migrationmonitor/settings.py]: `settings.py` are used as base settings file, where you can override it with `local_settings.py`.
So, you need either to change the logic of "Generate settings.py from template" in `ansible_installer/roles/monitor/tasks/main.yml` file or undo this change.

Comment (vladiskuz) [inline migrationmonitor/settings.py]: I think we should use only one configuration file, since the two configuration files will add confusion. If we use the migration-monitor for the first time it is not obvious that we can / should add local_settings.py and which of the two configuration files will be used as the main configuration file.

Comment (rk4n) [inline migrationmonitor/settings.py]: it's ok, just make it consistent with ansible scripts.

Comment (vladiskuz) [inline migrationmonitor/settings.py]: I've changed it https://github.com/rk4n/migration-monitor/pull/1/commits/97ced27a2149088fa2db7171c33bbef559d7f9e0#diff-e8f7df7790166d327e7acef54852fd92L19
